src/puppetlabs/puppetdb/http/command.clj

```diff
@@ -207,16 +212,21 @@
                            (update submit-params "version" str)
                            submit-params)
            ;; Replace read-body when our queue supports streaming
-           do-submit #(enqueue-fn (realize-body body max-command-size)
-                                  uuid
-                                  submit-params)]
+           do-submit (fn [command-callback]
+                       (enqueue-fn
+                        (get submit-params "command")
+                        (Integer/parseInt (get submit-params "version"))
+                        (get submit-params "certname")
+                        (stream-with-max-check body max-command-size)
+                        command-callback))]
+
        (if (some-> completion-timeout-ms pos?)
-         (blocking-submit-command do-submit (get-response-pub)
+         (blocking-submit-command do-submit
                                   uuid
                                   completion-timeout-ms)
          (do
-           (do-submit)
-           (http/json-response {:uuid uuid}))))
+           (do-submit identity)
```

Comment:
Currently all command refs include a callback. They will all be identity except in the blocking post case.